migrator/migrations/m_176_to_m_177_network_baselines_cidr/migration.go

```diff
@@ -0,0 +1,137 @@
+package m176tom177
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/migrator/migrations"
+	"github.com/stackrox/rox/migrator/migrations/m_176_to_m_177_network_baselines_cidr/networkbaselinestore"
+	"github.com/stackrox/rox/migrator/migrations/m_176_to_m_177_network_baselines_cidr/networkentitystore"
+	"github.com/stackrox/rox/migrator/types"
+	"github.com/stackrox/rox/pkg/postgres"
+	"gorm.io/gorm"
+)
+
+const (
+	startSeqNum = 175
+
+	batchSize = 500
+)
+
+var (
+	migration = types.Migration{
+		StartingSeqNum: startSeqNum,
+		VersionAfter:   &storage.Version{SeqNum: int32(startSeqNum + 1)},
+		Run: func(database *types.Databases) error {
+			return addCIDRBlockToBaselines(database.PostgresDB, database.GormDB)
+		},
+	}
+)
+
+func init() {
+	migrations.MustRegisterMigration(migration)
+}
+
+func updatePeer(
+	ctx context.Context,
+	networkEntityStore networkentitystore.Store,
+	deploymentId string,
+	peer *storage.NetworkBaselinePeer,
+) (bool, error) {
+	info := peer.GetEntity().GetInfo()
+	externalSource := info.GetExternalSource()
+	if externalSource != nil {
+		// Peer is an external entity and needs to be updated with CIDR block.
+		// We need to query data from the Network Entity store to fetch the CIDR block information
+		// using peer ID (i.e. external source ID) and append to the Network Entity property in
+		// the baseline.
+		entity, exists, err := networkEntityStore.Get(ctx, info.GetId())
+		if err != nil {
+			return false, err
+		}
+
+		if !exists {
+			return false, errors.Wrapf(err, "no network entity for peer %s in baseline for deployment %s",
+				info.GetId(),
+				deploymentId)
+		}
+
+		if entity.GetInfo().GetExternalSource() == nil {
+			return false, errors.Wrapf(err, "inconsistent type for peer %s in baseline for deployment %s: expecting EXTERNAL_SOURCE but is %s",
+				info.GetId(),
+				deploymentId,
+				entity.GetInfo().GetType())
+		}
+
+		entityCidrBlock := entity.GetInfo().GetExternalSource().GetCidr()
+		externalSource.Source = &storage.NetworkEntityInfo_ExternalSource_Cidr{
+			Cidr: entityCidrBlock,
+		}
+
+		peer.Entity.Info.Desc = &storage.NetworkEntityInfo_ExternalSource_{
+			ExternalSource: externalSource,
+		}
+		return true, nil
+	}
+	return false, nil
+}
+
+func addCIDRBlockToBaselines(postgresDB *postgres.DB, gormDB *gorm.DB) error {
+	ctx := context.Background()
+	networkBaselineStore := networkbaselinestore.CreateTableAndNewStore(ctx, postgresDB, gormDB)
```

Comment:
We are not changing the schemas so we do not need to use `CreateTableAndNewStore`.  We simply need to get a new store as the tables will have already been created and populated from all the previous migrations.

Reply:
That makes sense, thanks!